Re: [RFA/commit/testsuite] Add test for -file-list-exec-source-files
http://sourceware.org/ml/gdb-patches/2010-09/msg00090.html
http://sourceware.org/ml/gdb-cvs/2010-09/msg00019.html

### src/gdb/testsuite/ChangeLog	2010/08/31 18:40:09	1.2434
### src/gdb/testsuite/ChangeLog	2010/09/02 15:19:56	1.2435
## -1,3 +1,8 @@
+2010-09-02  Jan Kratochvil  <jan.kratochvil@redhat.com>
+            Joel Brobecker  <brobecker@adacore.com>
+
+	* gdb.dwarf/dw2-filename.S, gdb.dwarf/dw2-filename.exp: New files.
+
 2010-08-31  Doug Evans  <dje@google.com>
 
 	* gdb.dwarf2/dw4-sig-types.exp: Use .x for executable name.
--- src/gdb/testsuite/gdb.dwarf2/dw2-filename.S
+++ src/gdb/testsuite/gdb.dwarf2/dw2-filename.S	2011-07-26 21:39:34.094053000 +0000
@@ -0,0 +1,53 @@
+/* This testcase is part of GDB, the GNU debugger.
+
+   Copyright 2004, 2007, 2008, 2009, 2010 Free Software Foundation, Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.  */
+
+/* Debug information */
+
+	.section .debug_info
+.Lcu1_begin:
+	/* CU header */
+	.4byte	.Lcu1_end - .Lcu1_start		/* Length of Compilation Unit */
+.Lcu1_start:
+	.2byte	2				/* DWARF Version */
+	.4byte	.Labbrev1_begin			/* Offset into abbrev section */
+	.byte	4				/* Pointer size */
+
+	/* CU die */
+	.uleb128 1				/* Abbrev: DW_TAG_compile_unit */
+	.ascii	"file1.txt\0"			/* DW_AT_name */
+	.ascii	"GNU C 3.3.3\0"			/* DW_AT_producer */
+	.byte	1				/* DW_AT_language (C) */
+
+.Lcu1_end:
+
+/* Abbrev table */
+	.section .debug_abbrev
+.Labbrev1_begin:
+	.uleb128	1			/* Abbrev code */
+	.uleb128	0x11			/* DW_TAG_compile_unit */
+	.byte		0			/* has_children */
+	.uleb128	0x3			/* DW_AT_name */
+	.uleb128	0x8			/* DW_FORM_string */
+	.uleb128	0x25			/* DW_AT_producer */
+	.uleb128	0x8			/* DW_FORM_string */
+	.uleb128	0x13			/* DW_AT_language */
+	.uleb128	0xb			/* DW_FORM_data1 */
+	.byte		0x0			/* Terminator */
+	.byte		0x0			/* Terminator */
+
+	.byte		0x0			/* Terminator */
+	.byte		0x0			/* Terminator */
--- src/gdb/testsuite/gdb.dwarf2/dw2-filename.exp
+++ src/gdb/testsuite/gdb.dwarf2/dw2-filename.exp	2011-07-26 21:39:34.569595000 +0000
@@ -0,0 +1,44 @@
+# Copyright 2004, 2005, 2007, 2008, 2009, 2010 Free Software Foundation, Inc.
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Note: Inspired from dw2-basic.exp.
+
+# This test can only be run on targets which support DWARF-2 and use gas.
+# For now pick a sampling of likely targets.
+if {![istarget *-*-linux*]
+    && ![istarget *-*-gnu*]
+    && ![istarget *-*-elf*]
+    && ![istarget *-*-openbsd*]
+    && ![istarget arm-*-eabi*]
+    && ![istarget powerpc-*-eabi*]} {
+    return 0
+}
+
+set testfile "dw2-filename"
+set srcfile ${testfile}.S
+set binfile ${objdir}/${subdir}/${testfile}.x
+
+if  { [gdb_compile "${srcdir}/${subdir}/${srcfile}" "${binfile}" object {nodebug}] != "" } {
+    return -1
+}
+
+clean_restart ${testfile}.x
+
+# Verify that -file-list-exec-source-files returns a list where
+# the "file" is "file1.txt" (file base name), and "fullname" is
+# the full path to that file.  What we want to verify, most of all,
+# is that the file and fullname fields are now inverted.
+gdb_test "interpreter-exec mi -file-list-exec-source-files" \
+         ".*{file=\"file1\\.txt\",fullname=\".+file1\\.txt\"}.*"
